osutil/syncdir.go

```diff
@@ -63,16 +63,28 @@ var ErrSameState = fmt.Errorf("file state has not changed")
 // exhausted.
 //
 // In all cases, the function returns the first error it has encountered.
-func EnsureDirState(dir, glob string, content map[string]*FileState) (changed, removed []string, err error) {
-	if _, err := filepath.Match(glob, "foo"); err != nil {
-		panic(fmt.Sprintf("EnsureDirState got invalid pattern %q: %s", glob, err))
+func EnsureDirStateMany(dir string, globs []string, content map[string]*FileState) (changed, removed []string, err error) {
+	for _, glob := range globs {
+		if _, err := filepath.Match(glob, "foo"); err != nil {
```

Comment:
Oh, what is this? I realize it was in the existing code, but it looks perplexing.

Reply:
maybe this'll help: `filepath.Glob("expr")` returns the list of files in the directory for which `filepath.Match` is true

Reply:
Ah, now that I understand your question, the answer is that it's checking the globs syntax before actually doing any work

Reply:
Ok I see, makes sense, thanks!

Reply:
Worth a comment in the code.